From fe670d289e09d807fb760194290fbb35ab6230a5 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sat, 29 Jun 2019 17:06:24 +0800
Subject: [PATCH] Add dracut args for /home to bootloader

When /home is on a separate device like LVM or RAID dracut needs
to know about this so add the needed rd.* parameters to the cmdline.
This also works when upgrading.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/bootloader/base.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pyanaconda/bootloader/base.py b/pyanaconda/bootloader/base.py
index 0550745..d7506ac 100644
--- a/pyanaconda/bootloader/base.py
+++ b/pyanaconda/bootloader/base.py
@@ -729,6 +729,11 @@ class BootLoader(object):
         if usr_device:
             dracut_devices.extend([usr_device])
 
+        # Does /home have its own device? If so, we need to tell dracut
+        home_device = storage.mountpoints.get("/home")
+        if home_device:
+            dracut_devices.extend([home_device])
+
         netdevs = [d for d in storage.devices \
                    if (getattr(d, "complete", True) and
                        isinstance(d, NetworkStorageDevice))]
-- 
2.7.4

